CMake/GOMCManageMPI.cmake

```diff
@@ -0,0 +1,176 @@
+      # work.
+
+      find_program(MPIEXEC
+          NAMES mpiexec mpirun lamexec srun aprun poe
+          HINTS ${MPI_HOME} $ENV{MPI_HOME}
+          PATH_SUFFIXES bin
+          DOC "Executable for running MPI programs.")
+
+      set(MPIEXEC_NUMPROC_FLAG "-np" CACHE STRING "Flag used by MPI to specify the number of processes for MPIEXEC; the next option will be the number of processes.")
+      set(MPIEXEC_PREFLAGS     ""    CACHE STRING "These flags will be directly before the executable that is being run by MPIEXEC.")
+      set(MPIEXEC_POSTFLAGS    ""    CACHE STRING "These flags will come after all flags given to MPIEXEC.")
+      set(MPIEXEC_MAX_NUMPROCS "2"   CACHE STRING "Maximum number of processors available to run MPI applications.")
+      mark_as_advanced(MPIEXEC MPIEXEC_NUMPROC_FLAG MPIEXEC_PREFLAGS MPIEXEC_POSTFLAGS MPIEXEC_MAX_NUMPROCS)
+  endif()
+
+  if(MPI_FOUND)
+    include(${PROJECT_SOURCE_DIR}/CMake/GOMCTestMPI_IN_PLACE.cmake)
+    if (GOMC_MPI_IN_PLACE)
+      GOMC_test_mpi_in_place(MPI_IN_PLACE_EXISTS)
+    endif()
+
+    # Find path of the mpi compilers
+    if (${MPI_C_FOUND})
+        get_filename_component(_mpi_c_compiler_path "${MPI_C_COMPILER}" PATH)
+        get_filename_component(_mpiexec_path "${MPIEXEC}" PATH)
+    else()
+        get_filename_component(_cmake_c_compiler_path "${CMAKE_C_COMPILER}" PATH)
+        get_filename_component(_cmake_cxx_compiler_path "${CMAKE_CXX_COMPILER}" PATH)
+    endif()
+
+    # Test for and warn about unsuitable MPI versions
+    #
+    # Execute the ompi_info binary with the full path of the compiler wrapper
+    # found, otherwise we run the risk of false positives.
+    find_file(MPI_INFO_BIN ompi_info
+              HINTS ${_mpi_c_compiler_path} ${_mpiexec_path}
+                    ${_cmake_c_compiler_path} ${_cmake_cxx_compiler_path}
+              NO_DEFAULT_PATH
+              NO_SYSTEM_ENVIRONMENT_PATH
+              NO_CMAKE_SYSTEM_PATH)
+    if (MPI_INFO_BIN)
+      exec_program(${MPI_INFO_BIN}
+        ARGS -v ompi full
+        OUTPUT_VARIABLE OPENMPI_TYPE
+        RETURN_VALUE OPENMPI_EXEC_RETURN)
+      if(OPENMPI_EXEC_RETURN EQUAL 0)
+        string(REGEX REPLACE ".*Open MPI: \([0-9]+\\.[0-9]*\\.?[0-9]*\).*" "\\1" OPENMPI_VERSION ${OPENMPI_TYPE})
+        if(OPENMPI_VERSION VERSION_LESS "1.4.1")
+          MESSAGE(WARNING
+             "CMake found OpenMPI version ${OPENMPI_VERSION} on your system. "
+             "There are known problems with GROMACS and OpenMPI version < 1.4.1. "
+             "Please consider updating your OpenMPI if your MPI wrapper compilers "
+             "are using the above OpenMPI version.")
+        endif()
+        if(OPENMPI_VERSION VERSION_EQUAL "1.8.6")
+          MESSAGE(WARNING
+             "CMake found OpenMPI version ${OPENMPI_VERSION} on your system. "
+             "This OpenMPI version is known to leak memory with GROMACS,"
+             "please update to a more recent version. ")
```

Comment:
same here

Reply:
"This OpenMPI version is known to leak memory with GOMC,"